metricbeat/module/kubernetes/state_deployment/state_deployment_test.go

```diff
@@ -56,27 +56,27 @@ func TestEventMapping(t *testing.T) {
 		"hosts":      []string{server.URL},
 	}
 
-	f := mbtest.NewEventsFetcher(t, config)
-
-	events, err := f.Fetch()
-	assert.NoError(t, err)
+	f := mbtest.NewReportingMetricSetV2(t, config)
+	events, errs := mbtest.ReportingFetchV2(f)
+	if len(errs) > 0 {
+		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+	}
+	assert.NotEmpty(t, events)
 
 	assert.Equal(t, 5, len(events), "Wrong number of returned events")
 
 	testCases := testCases()
 	for _, event := range events {
-		name, err := event.GetValue("name")
+		metricsetFields := event.MetricSetFields
+		name, err := metricsetFields.GetValue("name")
 		if err == nil {
```

Comment:
just curious, do we need to deal with when `err != nil` here?

Reply:
It was originally like this so I prefer to leave it